### ISSUE TITLE
Refactor: Move metric computation out of src/viz/ (computation decoupling)

### ISSUE THREAD

**Issue by petteriTeikari** [petteriTeikari/foundation_PLR#13]
## Summary

The `/code-qa` discovery scan (Iteration 1) found **12 CRITICAL computation decoupling violations** in `src/viz/`. Per CRITICAL-FAILURE-003, visualization code must ONLY read pre-computed metrics from DuckDB — it must NEVER compute metrics itself.

### Active computation (no deprecation marker):
- `src/viz/dca_plot.py:37` — Defines `compute_net_benefit()` as a public export
- `src/viz/dca_plot.py:440,532` — Imports `decision_curve_analysis` from `src.stats.clinical_utility`
- `src/viz/stratos_figures.py:514` — Uses `compute_net_benefit` to compute DCA curves on-the-fly

### Deprecated but still present:
- `src/viz/retained_metric.py:75` — `sklearn.metrics` (brier, auroc)
- `src/viz/prob_distribution.py:31` — `sklearn.metrics` (auroc)
- `src/viz/metric_vs_cohort.py:28` — `sklearn.metrics` (brier, auroc)
- `src/viz/metric_registry.py:175+` — `sklearn.metrics` (auroc, brier, f1) marked "extraction use only"
- `src/viz/calibration_plot.py:643,725` — `stats.calibration_extended` (noqa markers)
- `src/viz/fig_instability_plots.py:23` — `src.stats.pminternal_analysis`
- `src/viz/generate_instability_figures.py:42` — `src.stats.pminternal_analysis`

## Required Architecture

```
[Extraction Block]                    [Visualization Block]
  MLflow → compute metrics → DuckDB    DuckDB → read-only → Figures
  ALL computation here                  NO computation
```

## Proposed Fix (per file)

| File | Action |
|------|--------|
| `dca_plot.py` | Pre-compute DCA curves in extraction, store in DuckDB as `dca_curves` table. Viz reads from DB. |
| `retained_metric.py` | Pre-compute retention curves at various rates during extraction. Viz reads from DB. |
| `prob_distribution.py` | Pre-compute distribution stats during extraction. Remove sklearn import. |
| `metric_vs_cohort.py` | Pre-compute cohort metrics during extraction. Remove sklearn import. |
| `metric_registry.py` | Move sklearn-dependent metric definitions to `src/stats/` or `src/extraction/`. |
| `calibration_plot.py` | Remove deprecated functions, ensure calibration metrics come from DuckDB. |
| `fig_instability_plots.py` | Pre-compute instability data in extraction, pass as data input. |

## Scope

This is a large refactor touching the extraction pipeline and multiple visualization modules. Each file should be migrated independently with its own PR/commit to minimize risk.

## Context

- Discovered by: `/code-qa` skill, Iteration 1
- Rule: CRITICAL-FAILURE-003 (computation decoupling)
- See: `.claude/docs/meta-learnings/CRITICAL-FAILURE-003-computation-decoupling-violation.md`
- Related: `.claude/skills/code-qa/state/qa-state.json` tracks all findings

## Labels

`refactor`, `code-quality`, `computation-decoupling`
